qa/tasks/cephfs/test_exports.py

```diff
@@ -175,3 +175,131 @@ def test_session_race(self):
         # Check if rank1 changed (standby tookover?)
         new_rank1 = self.fs.get_rank(rank=1)
         self.assertEqual(rank1['gid'], new_rank1['gid'])
+
+class TestKillPoints(CephFSTestCase):
+    # Two active MDS
+    MDSS_REQUIRED = 2
+
+    init = False
+
+    def _setup_cluster(self):
+        # Set Multi-MDS cluster
+        self.fs.set_max_mds(2)
+
+        self.fs.wait_for_daemons()
+        
+        all_daemons = self.fs.get_daemon_names()
+
+        # Create test data
+        if self.init is False:
+            self._populate_data(8)
+            self.init = True
+
+        return True
+
+    def _populate_data(self, nfiles):
+            size_mb = 8
+            dname = "abc"
+            self.mount_a.run_shell(["mkdir", dname])
+            for i in range(nfiles):
+                fname =dname.join("/").join(str(i)).join(".txt")
+                self.mount_a.write_n_mb(fname, size_mb)
+
+            # Get the list of file to confirm the count of files
+            self.org_files = self.mount_a.ls()
+
+    def _verify_data(self):
+        s1 = set(self.mount_a.ls())
+        s2 = set(self.org_files)
+
+        if s1.isdisjoint(s2):
+            log.error("Directory contents org: %s" %str(org_files))
+            log.error("Directory contents new: %s" %str(out))
+            return False
+
+        log.info("Directory contents matches")
+        return True
+
+    def _run_export_dir(self, importv, exportv):
+        # Wait till all MDS becomes active
+        self.fs.wait_for_daemons()
+
+        # Get all active ranks
+        ranks = self.fs.get_all_mds_rank()
```

Comment:
better is:

`status = self.fs.status()`

so that your test is only doing `fs dump` once. Then you can pass that status to other methods like `self.fs.get_rank(status=status)`.

Reply:
(Also, you're not using this `ranks` variable.)